### ISSUE TITLE
Cypress Grep typescript definition for describe tags

### ISSUE THREAD

**Issue by danreale** [cypress-io/cypress#25026]
### Current behavior

When trying to add tags to my describe block like in the example 
`describe('block with config tag', { tags: '@smoke' }, () => {})`
 I am seeing red lines.  The error that comes back is 

```
Argument of type '{ tags: string; }' is not assignable to parameter of type 'SuiteConfigOverrides'.
  Object literal may only specify known properties, and 'tags' does not exist in type 'SuiteConfigOverrides'.
```

It blocks are working as expected

### Desired behavior

support/typescript support for describe block with cypress grep as stated in the examples

### Test code to reproduce

Install plugin as stated in the instructions and try to add a describe block with tags in VS code

### Cypress Version

12.0.1

### Node version

16.13.0

### Operating System

mac 10.15.7 (Catalina)

**Comment by vergjor** [comment]
@danreale you could add the following to suppress the error:

```
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
describe('block with config tag', { tags: '@smoke' }, () => {})
```

**Comment by danreale** [comment]
Thanks @vergjor.

**Comment by LordRhys** [comment]
@vergjor will this still allow use of grepping tags, we use this to tell whether that script can be run in a certain environment, and noticed this error marking in VS Code when we switched to Cypress 11

**Comment by vergjor** [comment]
@LordRhys Yup, I had the same issue with @cypress/grep and using these comments helped. It just suppressed the error and the tags work. The annoying this however is that you need to add it before each test that has tags

**Comment by LordRhys** [comment]
@vergjor thanks, we'll be testing it on Monday

**Comment by teofilomonteiro** [comment]
> @danreale you could add the following to suppress the error:
> 
> ```
> // eslint-disable-next-line @typescript-eslint/ban-ts-comment
> // @ts-ignore
> describe('block with config tag', { tags: '@smoke' }, () => {})
> ```

please don't do that... it is just a way to quickly get more errors in the future.

I think the right solution is to add to your tsconfig.json into types the @cypress/grep `"types": ["cypress", "node", "@cypress/grep"]`. this will solve all your issues without suppressing them. like described here https://github.com/cypress-io/cypress/tree/develop/npm/grep#typescript-support

**Comment by AnnemarieD** [comment]
We are experiencing the same error after upgrading @cypress/grep to v5.0.0 or higher. In v4.X it doesn't show this message.
The weird thing is, if I click on `SuiteConfigOverrides` in IntelliJ it shows the property `tags?: string | string[]` (depending on what I have declared in my test it shows `string` or `string[]`)

We updated grep as part of upgrading NX to v20.5, Angular to 19.2 and Cypress to 14.2.0. 

The error is
```
Argument of type '{ tags: string; }' is not assignable to parameter of type 'SuiteConfigOverrides'.
  Object literal may only specify known properties, and 'tags' does not exist in type 'SuiteConfigOverrides'.
```

Workaround is to declare the properties for `SuiteConfigOverrides` ourselves, but this feels weird when both IntelliJ and myself can find the expected typing easily, only Typescript can't. We checked the setup, there seems to be nothing wrong. 

Not sure how to fix this, we might have to go for the workaround as going down a major version is not what we or NX want.
We already have the types declared in tsconfig.json as described above and I dont want to ignore this ts rule

**Comment by AnnemarieD** [comment]
We solved this by adding `/// <reference types="@cypress/grep/dist/index.d.ts" />` to the top of `types.d.ts` file and removing `"@cypress/grep"` from types in `tsconfig.json`
